### PR TITLE
database: mask mysql password in JSON marshaling

### PR DESCRIPTION
**Pull request by adamdecaf**
This is useful to hide passwords with `GET /config` on admin servers.

### CODE REVIEW

File: database/model_config.go [hunk 1]
```diff
@@ -15,6 +18,21 @@ type MySQLConfig struct {
...
+	type Aux struct {
+		Address     string
+		User        string
+		Password    string
+		Connections ConnectionsConfig
+	}
+	return json.Marshal(Aux{
+		Address:     m.Address,
+		User:        m.User,
+		Password:    "a***z",
```
Comment (vxio) [inline database/model_config.go:31]: I think this needs to be `mask.Password(..)`

**Review by vxio — APPROVED** [review]

File: mask/password.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	"unicode/utf8"
+)
+
+func Password(s string) string {
+	if utf8.RuneCountInString(s) < 3 {
+		return "**" // too short, we can't mask anything
+	} else {
+		// turn 'password' into 'p******d'
+		first, last := s[0:1], s[len(s)-1:]
+		return fmt.Sprintf("%s%s%s", first, strings.Repeat("*", len(s)-2), last)
```
Comment (atonks2) [inline mask/password.go:19]: Does it matter that this exposes the first and last chars, and the length of the password?

Comment (adamdecaf) [reply mask/password.go:19]: This is designed for admin servers of applications and to give operators some level of visibility to ensure the password is correct without exposing the credentials. It's a tradeoff and currently applications would render the entire password.

Comment (atonks2) [reply mask/password.go:19]: I wonder if we could use a fixed amount of `*`? That'd still be enough for the admin to recognize the first and last chars, but not leak the password length.

Comment (adamdecaf) [reply mask/password.go:19]: Yea we can do that. 5 stars?

Comment (atonks2) [reply mask/password.go:19]: I think that sounds good.

**Review by atonks2 — APPROVED** [review]

**Pull request merged**